src/main/java/org/jruby/ext/openssl/ASN1.java

```diff
@@ -1071,15 +1071,14 @@ else if ( obj instanceof ASN1GraphicString ) {
                     break;
             }
 
-            if (taggedObj.getTagClass() == BERTags.APPLICATION) {
+            try {
                 final ASN1Sequence sequence = (ASN1Sequence) taggedObj.getBaseUniversal(false, SEQUENCE);
                 @SuppressWarnings("unchecked")
                 final RubyArray valArr = decodeObjects(context, ASN1, sequence.getObjects());
                 return ASN1.getClass("ASN1Data").newInstance(context, new IRubyObject[] { valArr, tag, tag_class }, Block.NULL_BLOCK);
-            } else {
-                IRubyObject val = decodeObject(context, ASN1, taggedObj.getBaseObject());
-                final RubyArray valArr = runtime.newArray(val);
-                return ASN1.getClass("ASN1Data").newInstance(context, new IRubyObject[] { valArr, tag, tag_class }, Block.NULL_BLOCK);
+            } catch (IllegalStateException e) {
```

Comment:
seems a bit unusual, is there a reason why we have to resort to the exception as flow control?

Reply:
I couldn't make the decoding of `"@\x03bla"` work in any other way (see the respective unit test, it fails with " unexpected implicit primitive encoding" without this). Do you know of a better alternative?

Reply:
okay, thanks - there does not seem to be a way API to check (e.g. `isParsed()`) isn't public :disappointed: